hazelcast/src/main/java/com/hazelcast/spi/properties/ClusterProperty.java

```diff
@@ -1546,6 +1546,18 @@ private int getWhenNoSSLDetected() {
             = new HazelcastProperty("hazelcast.serialization.version",
             BuildInfoProvider.getBuildInfo().getSerializationVersion());
 
+    /**
+     * Property allows override of default Java serializers for backward compatibility.
+     *
+     * For example application using older version of Hazelcast provides custom java.util.Optional serializer.
+     * Since version 4.2 of Hazelcast registration of any serializer for an embedded type is not possible
+     * until this flag is set to true.
+     *
+     * @since 4.2
+     */
+    public static final HazelcastProperty SERIALIZATION_ALLOW_OVERRIDE_DEFAULT_SERIALIZERS
```

Comment:
Hi @sgflt,
since we have decided to give full config support for this the property support is needed. Can we remove this back? Sorry for the inconvenience.

Reply:
Hi @sancar, I followed widely used way of configuration:
1) system property - useful for debug or fine tuning, overrides config file
2) standard config file as persistent configuration

So the final decision is to drop the system property support? It simplifies some code paths, but removes duality of configuration, which may be useful in some use cases.

Reply:
We are not putting configuration to both properties and config unless we have to. 
And I have just learned that we applied a feature that allows you to override Config via properties anyway. 
https://docs.hazelcast.org/docs/latest/manual/html-single/#overriding-configuration
So when you add it to Config, the property support is added automagically :)

Reply:
Oh nice :) I am going to investigate this feature.

Reply:
Property removed